bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/live/TelldusLiveDeviceController.java

```diff
@@ -276,37 +285,66 @@ public void onRequest(TellstickDeviceEvent newDevices) {
+        LocalDateTime start = LocalDateTime.now();
         T resultObj = null;
         try {
-            for (int i = 0; i < MAX_RETRIES; i++) {
-                try {
-                    resultObj = innerCallRest(uri, response);
-                    break;
-                } catch (TimeoutException e) {
-                    logger.warn("TimeoutException error in get", e);
-                } catch (InterruptedException e) {
-                    logger.warn("InterruptedException error in get", e);
-                }
-            }
-        } catch (JAXBException e) {
-            logger.warn("Encoding error in get", e);
+            resultObj = innerCallRest(uri, response);
+        } catch (TimeoutException | InterruptedException | ExecutionException | JAXBException | XMLStreamException e) {
             logResponse(uri, e);
-            throw new TelldusLiveException(e);
-        } catch (XMLStreamException e) {
-            logger.warn("Communication error in get", e);
-            logResponse(uri, e);
-            throw new TelldusLiveException(e);
-        } catch (ExecutionException e) {
-            logger.warn("ExecutionException error in get", e);
+            monitorAdditionalRequest(start, LocalDateTime.now(), e);
             throw new TelldusLiveException(e);
         }
+        monitorAdditionalRequest(start, LocalDateTime.now(), null);
         return resultObj;
     }
 
+    private void monitorAdditionalRequest(LocalDateTime start, LocalDateTime end, @Nullable Throwable exception) {
+        if (!logger.isDebugEnabled()) {
+            return;
+        }
+        long duration = ChronoUnit.MILLIS.between(start, end);
+        sumRequestDuration += duration;
+        nbRequest++;
+        if (duration < minRequestDuration) {
+            minRequestDuration = duration;
+        }
+        if (duration > maxRequestDuration) {
+            maxRequestDuration = duration;
+        }
+        if (exception instanceof TimeoutException) {
+            nbTimeouts++;
+        } else if (exception != null) {
+            nbErrors++;
+        }
+    }
+
+    public long getAverageRequestDuration() {
+        return nbRequest == 0 ? 0 : sumRequestDuration / nbRequest;
+    }
+
+    public long getMinRequestDuration() {
+        return minRequestDuration;
+    }
+
+    public long getMaxRequestDuration() {
+        return maxRequestDuration;
+    }
+
+    public int getNbTimeouts() {
+        return nbTimeouts;
+    }
+
+    public int getNbErrors() {
+        return nbErrors;
+    }
+
     private <T> T innerCallRest(String uri, Class<T> response) throws InterruptedException, ExecutionException,
             TimeoutException, JAXBException, FactoryConfigurationError, XMLStreamException {
         Future<Response> future = client.prepareGet(uri).execute();
         Response resp = future.get(REQUEST_TIMEOUT_MS, TimeUnit.MILLISECONDS);
+        if (resp.getStatusCode() != 200) {
```

Comment:
Consider `HttpStatus.OK_200` or similar (too lazy to check right now if response if from Jetty). Could also be stored in local variable to avoid additional call in next line.

Reply:
Unfortunately, the binding does not use Jetty HTTP client. I don't want to include Jetty just for this constant.